### PR TITLE
sles4sap/publiccloud: avoid extra 'wait_for_ssh' in 'stop_hana'

### PR DESCRIPTION
**Pull request by zhaocongjia**
In 'stop_hana', for the 'crash' test scenario, there's no need to run 'wait_for_ssh' because we already have 'wait_hana_node_up'.

This is a follow up of https://github.com/os-autoinst/os-autoinst-distri-opensuse/pull/20539.

- Related ticket: https://jira.suse.com/browse/TEAM-9823
- Needles: none
- Verification run:
msi:
https://openqaworker15.qa.suse.cz/tests/302517 (fail at different place)
https://openqaworker15.qa.suse.cz/tests/302518
https://openqaworker15.qa.suse.cz/tests/302519
https://openqaworker15.qa.suse.cz/tests/302520
sbd:
https://openqaworker15.qa.suse.cz/tests/302587 (fail at different place)
https://openqaworker15.qa.suse.cz/tests/302588
https://openqaworker15.qa.suse.cz/tests/302589
https://openqaworker15.qa.suse.cz/tests/302590
spn:
https://openqaworker15.qa.suse.cz/tests/302591
https://openqaworker15.qa.suse.cz/tests/302592
https://openqaworker15.qa.suse.cz/tests/302593
https://openqaworker15.qa.suse.cz/tests/302594

### CODE REVIEW

**Review by lpalovsky — APPROVED** [review]
LGTM

File: lib/sles4sap_publiccloud.pm [hunk 1]
```diff
@@ -434,6 +434,7 @@ sub stop_hana {
     else {
         my $sapadmin = lc(get_required_var('INSTANCE_SID')) . 'adm';
         $self->run_cmd(cmd => $cmd, runas => $sapadmin, timeout => $timeout);
+        $self->{my_instance}->wait_for_ssh(username => 'cloudadmin', scan_ssh_host_key => 1);
```
Comment (mpagot) [inline lib/sles4sap_publiccloud.pm:437]: Is this one intended to add a `wait_for_ssh` in `stop` and `kill` scenario?
What is the purpose of this change?

File: t/12_sles4sap_publicccloud.t [hunk 2]
```diff
@@ -177,6 +177,9 @@ subtest "[stop_hana]" => sub {
             return; }
     );
     my $self = sles4sap_publiccloud->new();
+    my $mock_pc = Test::MockObject->new();
```
Comment (mpagot) [inline t/12_sles4sap_publicccloud.t:180]: Not sure which one is better but here:
- https://github.com/os-autoinst/os-autoinst-distri-opensuse/pull/18937/files#diff-19cb97f4561f758bdb619417961e20a06b7f877f46f3ccc27be414f0ae54a3afR23
- https://github.com/os-autoinst/os-autoinst-distri-opensuse/pull/18937/files#diff-19cb97f4561f758bdb619417961e20a06b7f877f46f3ccc27be414f0ae54a3afR118

I did it differently. I think that both implementation are achieving the same: 
- mock the wait_for_ssh, avoiding the real implementation to be invoked
- not testing anything about it: so both implementation does not later check anything about the mock execution (does anyone call it? how many time? ...)

**Review by mpagot — APPROVED** [review]
LGTM

File: lib/sles4sap_publiccloud.pm (hunk 1 above)
Comment (mpagot) [inline lib/sles4sap_publiccloud.pm:437]: There are other two calls to `wait_for_ssh` in this same file, they both specify explicit `timeout`. I think it make sense to do the same here too.
